programs/.gitignore

```diff
@@ -45,6 +45,7 @@ ssl/mini_client
 test/benchmark
 test/ecp-bench
 test/selftest
+test/header_test
```

Comment:
I don't think `header_test` really conveys what we want to do here. Can we call it something like `cpp_dummy_build`? Or at least something that says it's a C++ test.

Reply:
Done - renamed to cpp_dummy_build.